### PR TITLE
[Network] Fix Network Watcher commands

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes the missing client issue for certain Network Watcher commands.  Restores network watcher tests that would have prevented the regression.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [X] Each command and parameter has a meaningful description.
- [X] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5441
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py [hunk 1]
```diff
@@ -397,112 +396,112 @@ def get_virtual_network_validator(has_type_field=False, allow_none=False, allow_
...
 
 # COMMAND NAMESPACE VALIDATORS
 
-def process_ag_listener_create_namespace(namespace):  # pylint: disable=unused-argument
+def process_ag_listener_create_namespace(cmd, namespace):  # pylint: disable=unused-argument
     from msrestazure.tools import is_valid_resource_id
     if namespace.frontend_ip and not is_valid_resource_id(namespace.frontend_ip):
         namespace.frontend_ip = _generate_ag_subproperty_id(
-            namespace, 'frontendIpConfigurations', namespace.frontend_ip)
+            cmd.cli_ctx, namespace, 'frontendIpConfigurations', namespace.frontend_ip)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py]: Am I under the right assumption that a few app-gateway/lb commands are not working right now? If yes, I suggest either unit test validators or add scenario tests with SDK's create method mock'd out to be a non-op.

File: src/azure-cli-core/azure/cli/core/commands/template_create.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def get_folded_parameter_validator(
         raise CLIError('Cannot use default_none=True if allow_none=False')
 
     # construct the validator
-    def validator(namespace):
+    def validator(cmd, namespace):  # pylint: disable=unused-argument
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/commands/template_create.py]: Is this change needed, as I don't see any usages  in the body.  Or at the line 113, we should the update `namespace.cmd.cli_ctx`?

Comment (tjprescott) [reply src/azure-cli-core/azure/cli/core/commands/template_create.py]: It is necessary as I tried to remove it do get rid of the pylint warning and there were a billion errors.  I will fix up the namespace.cmd.cli_ctx reference. Thanks for catching it!

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py (hunk 1 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py]: They were working because they were relying on `namespace.cmd.cli_ctx`.  They are covered by scenario tests. I wanted to update the syntax across the board to use the recommended signature instead of that old syntax.
